src/librustc_parse/parser/generics.rs

```diff
@@ -56,12 +56,14 @@ impl<'a> Parser<'a> {
 
         self.sess.gated_spans.gate(sym::min_const_generics, const_span.to(self.prev_token.span));
 
+        let default = if self.eat(&token::Eq) { Some(self.parse_expr()?) } else { None };
```

Comment:
We can't really allow arbitrary expressions here for the same reason that const args require braces,

`struct Foo<const N: usize = 7>`... is otherwise parsed as an expression / requires infinite lookahead.

I think the correct way is to allow exactly what's currently allowed as const arguments (identifiers, literals and blocks).